### PR TITLE
r/aws_ce_cost_category Add tags to cost category

### LINKED ISSUE #25422: aws_ce_cost_category: add support for tagging categories
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

aws_ce_cost_category supports tagging cost categories

### New or Affected Resource(s)

* aws_ce_cost_category

### References
https://github.com/aws/aws-sdk-go-v2/blob/service/costexplorer/v1.19.0/service/costexplorer/api_op_CreateCostCategoryDefinition.go#L79

### PR DESCRIPTION
**Pull request by danquack**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #25422

Output from acceptance testing:

```
$ make testacc TESTS=TestAccCECostCategory_ PKG=ce
TF_ACC=1 go test ./internal/service/ce/... -v -count 1 -parallel 20 -run='TestAccCECostCategory_'  -timeout 180m
=== RUN   TestAccCECostCategory_tagRemove
=== PAUSE TestAccCECostCategory_tagRemove
=== RUN   TestAccCECostCategory_tagAdd
=== PAUSE TestAccCECostCategory_tagAdd
=== CONT  TestAccCECostCategory_tagRemove
=== CONT  TestAccCECostCategory_tagAdd
--- PASS: TestAccCECostCategory_tagRemove (36.02s)
--- PASS: TestAccCECostCategory_tagAdd (36.98s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ce	38.812s
```

### CODE REVIEW

File: internal/service/ce/cost_category.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
```
Comment (drewmullen) [inline internal/service/ce/cost_category.go:17]: I think you're missing the `CustomizeDiff`
```suggestion
	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
	"github.com/hashicorp/terraform-provider-aws/names"
)
func ResourceCostCategory() *schema.Resource {
	return &schema.Resource{
		CreateContext: resourceCostCategoryCreate,
		ReadContext:   resourceCostCategoryRead,
		UpdateContext: resourceCostCategoryUpdate,
		DeleteContext: resourceCostCategoryDelete,
		Importer: &schema.ResourceImporter{
			StateContext: schema.ImportStatePassthroughContext,
		},
                CustomizeDiff: customdiff.Sequence(verify.SetTagsDiff),
```

File: internal/service/ce/cost_category.go [hunk 2]
```diff
@@ -100,6 +101,7 @@ func ResourceCostCategory() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringLenBetween(0, 100),
 			},
+			"tags": tftags.TagsSchemaComputed(),
```
Comment (drewmullen) [inline internal/service/ce/cost_category.go:104]:
```suggestion
			"tags":     tftags.TagsSchema(),
			"tags_all": tftags.TagsSchemaComputed(),
```

**Comment by danquack** [comment]
Its unclear what the test name should be. I had it as `testAccCostCategoryConfig_WithTag` and semgrep told me it should be `testAcc<Resource>Config_<testDetail>` But when I change to `testAccCECostCategoryConfig_WithTag` it says `Do not use "CE" in func name inside ce package`. The test themselves pass, it is only a semgrep build error.

File: internal/service/ce/cost_category_test.go [hunk 3]
```diff
@@ -112,6 +112,74 @@ func TestAccCECostCategory_splitCharge(t *testing.T) {
...
+	rName := sdkacctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckCostCategoryDestroy,
+		ErrorCheck:        acctest.ErrorCheck(t, costexplorer.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCECostCategoryConfig_WithoutTag(rName),
```
Comment (YakDriver) [inline internal/service/ce/cost_category_test.go:161]:
```suggestion
				Config: testAccCostCategoryConfig_noTag(rName),
```

File: internal/service/ce/cost_category_test.go [hunk 4]
```diff
@@ -112,6 +112,74 @@ func TestAccCECostCategory_splitCharge(t *testing.T) {
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCostCategoryExists(resourceName, &output),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+			{
+				Config: testAccCECostCategoryConfig_WithTag(rName),
```
Comment (YakDriver) [inline internal/service/ce/cost_category_test.go:168]:
```suggestion
				Config: testAccCostCategoryConfig_tag(rName),
```

File: internal/service/ce/cost_category_test.go [hunk 5]
```diff
@@ -375,3 +443,52 @@ resource "aws_ce_cost_category" "test" {
 }
 `, rName, method)
 }
+
+func testAccCECostCategoryConfig_WithTag(rName string) string {
```
Comment (YakDriver) [inline internal/service/ce/cost_category_test.go:447]:
```suggestion
func testAccCostCategoryConfig_tag(rName string) string {
```

File: internal/service/ce/cost_category_test.go [hunk 6]
```diff
@@ -375,3 +443,52 @@ resource "aws_ce_cost_category" "test" {
...
+    type = "REGULAR"
+  }
+  tags = {
+    Name = %[1]q
+  }
+}
+`, rName)
+}
+
+func testAccCECostCategoryConfig_WithoutTag(rName string) string {
```
Comment (YakDriver) [inline internal/service/ce/cost_category_test.go:473]:
```suggestion
func testAccCostCategoryConfig_noTag(rName string) string {
```

**Review by drewmullen — APPROVED** [review]
LGTM! Nice work! 🚀

**Comment by jjshoe** [comment]
@YakDriver can I get you to review this when you get a second? Should be good to go.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccCECostCategory' PKG=ce ACCTEST_PARALLELISM=3 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ce/... -v -count 1 -parallel 3  -run=TestAccCECostCategory -timeout 180m
=== RUN   TestAccCECostCategoryDataSource_basic
=== PAUSE TestAccCECostCategoryDataSource_basic
=== RUN   TestAccCECostCategory_basic
=== PAUSE TestAccCECostCategory_basic
=== RUN   TestAccCECostCategory_disappears
=== PAUSE TestAccCECostCategory_disappears
=== RUN   TestAccCECostCategory_complete
=== PAUSE TestAccCECostCategory_complete
=== RUN   TestAccCECostCategory_splitCharge
=== PAUSE TestAccCECostCategory_splitCharge
=== RUN   TestAccCECostCategory_tags
=== PAUSE TestAccCECostCategory_tags
=== CONT  TestAccCECostCategoryDataSource_basic
=== CONT  TestAccCECostCategory_tags
=== CONT  TestAccCECostCategory_complete
--- PASS: TestAccCECostCategoryDataSource_basic (23.96s)
=== CONT  TestAccCECostCategory_disappears
--- PASS: TestAccCECostCategory_disappears (16.15s)
=== CONT  TestAccCECostCategory_basic
--- PASS: TestAccCECostCategory_complete (43.84s)
=== CONT  TestAccCECostCategory_splitCharge
--- PASS: TestAccCECostCategory_tags (57.21s)
--- PASS: TestAccCECostCategory_basic (23.35s)
--- PASS: TestAccCECostCategory_splitCharge (39.84s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ce	91.494s
```

**Comment by ewbankkit** [comment]
@danquack Thanks for the contribution :tada: :clap:.

**Comment by ewbankkit** [comment]
```console
% make providerlint golangci-lint
==> Checking source code with providerlint...
==> Checking source code with golangci-lint...
```

**Pull request merged**